apps/server/src/controllers/userControllers/getUserPersonalSavingsWallet.ts

```diff
@@ -0,0 +1,53 @@
+import { Response } from 'express'
+import { Op } from 'sequelize'
+import { HTTP_STATUS_CODE } from '../../constants'
+import { RequestExt } from '../../middlware/authorization/authentication'
+import Wallets, { WalletType } from '../../models/wallets'
+
+export const getUserPersonalSavingsWallet = async (
+  req: RequestExt,
+  res: Response
+) => {
+  try {
+    const { _userId: userId } = req.body
+    const personalSavingsWallet = await Wallets.findOne({
+      where: {
+        [Op.and]: [{ userId: userId }, { type: WalletType.SAVINGS }],
+      },
+    })
+    const personalGroupSavingsWallet = await Wallets.findOne({
+      where: {
+        [Op.and]: [{ userId: userId }, { type: WalletType.GROUP_WALLET }],
+      },
+    })
+    const personalGlobalWallet = await Wallets.findOne({
+      where: {
+        [Op.and]: [{ userId: userId }, { type: WalletType.GLOBAL }],
+      },
+    })
```

Comment:
i think this can be simplified by adding an OR, what do you think?

```
const personalWallets = await Wallets.findAll({
  where: {
    [Op.and]: [
      { userId: userId },
      {
        [Op.or]: [
          { type: WalletType.SAVINGS },
          { type: WalletType.GROUP_WALLET },
          { type: WalletType.GLOBAL }
        ]
      }
    ]
  }
});


```

Reply:
Alternatively, we could also have a wallets helper that we call for each Wallet  instance:

const findWallet = async (userId, walletType) => {
  return await Wallets.findOne({
    where: {
      [Op.and]: [{ userId: userId }, { type: walletType }],
    },
  })
}

Reply:
> Alternatively, we could also have a wallets helper that we call for each Wallet instance:
> 
> const findWallet = async (userId, walletType) => { return await Wallets.findOne({ where: { [Op.and]: [{ userId: userId }, { type: walletType }], }, }) }

This is actually more simplified with the helper function.

Reply:
> i think this can be simplified by adding an OR, what do you think?
> 
> ```
> const personalWallets = await Wallets.findAll({
>   where: {
>     [Op.and]: [
>       { userId: userId },
>       {
>         [Op.or]: [
>           { type: WalletType.SAVINGS },
>           { type: WalletType.GROUP_WALLET },
>           { type: WalletType.GLOBAL }
>         ]
>       }
>     ]
>   }
> });
> ```
> 
> FYI: I did not test this.

This works boss.

Reply:
> > Alternatively, we could also have a wallets helper that we call for each Wallet instance:
> > const findWallet = async (userId, walletType) => { return await Wallets.findOne({ where: { [Op.and]: [{ userId: userId }, { type: walletType }], }, }) }
> 
> Yeah, this is a good helper function, but will now take a longer process shaa, in a way. With this helper function, you can now use it in your file to fetch all the wallets as follows:
> 
> ```
> const findWalletsByType = async (userId) => {
>  const savingsWallet = await findWallet(userId, WalletType.SAVINGS);
>  const groupWallet = await findWallet(userId, WalletType.GROUP_WALLET);
>  const globalWallet = await findWallet(userId, WalletType.GLOBAL);
> 
>  return {
>    savings: savingsWallet,
>    group: groupWallet,
>    global: globalWallet,
>  };
> };
> 
> const walletResults = await findWalletsByType(userId);
> 
> // You can now access individual wallet types as needed
> const savingsWalletResult = walletResults.savings;
> const groupWalletResult = walletResults.group;
> const globalWalletResult = walletResults.global;
> ```

Just tested this code now. Works perfectly fine.

Reply:
> Alternatively, we could also have a wallets helper that we call for each Wallet instance:
> 
> const findWallet = async (userId, walletType) => { return await Wallets.findOne({ where: { [Op.and]: [{ userId: userId }, { type: walletType }], }, }) }

while this might work, it is not efficient for this scenario because you are making multiple calls on the DB.

Reply:
> > Alternatively, we could also have a wallets helper that we call for each Wallet instance:
> > const findWallet = async (userId, walletType) => { return await Wallets.findOne({ where: { [Op.and]: [{ userId: userId }, { type: walletType }], }, }) }
> 
> Yeah, this is a good helper function, but will now take a longer process shaa, in a way. With this helper function, you can now use it in your file to fetch all the wallets as follows:
> 
> ```
> const findWalletsByType = async (userId) => {
>  const savingsWallet = await findWallet(userId, WalletType.SAVINGS);
>  const groupWallet = await findWallet(userId, WalletType.GROUP_WALLET);
>  const globalWallet = await findWallet(userId, WalletType.GLOBAL);
> 
>  return {
>    savings: savingsWallet,
>    group: groupWallet,
>    global: globalWallet,
>  };
> };
> 
> const walletResults = await findWalletsByType(userId);
> 
> // You can now access individual wallet types as needed
> const savingsWalletResult = walletResults.savings;
> const groupWalletResult = walletResults.group;
> const globalWalletResult = walletResults.global;
> ```

while this might work, it is not efficient for this scenario because you are making multiple calls on the DB.

Reply:
> i think this can be simplified by adding an OR, what do you think?
> 
> ```
> const personalWallets = await Wallets.findAll({
>   where: {
>     [Op.and]: [
>       { userId: userId },
>       {
>         [Op.or]: [
>           { type: WalletType.SAVINGS },
>           { type: WalletType.GROUP_WALLET },
>           { type: WalletType.GLOBAL }
>         ]
>       }
>     ]
>   }
> });
> ```
> 
> FYI: I did not test this.

In order to implement the above i discovered that it was fetching the groups wallet as well. This is because userId assigned to the group wallet is that of the user that created the group instead of the groupId of of that group. So I made a few changes to wallet model, and the groupController in order to resolve this issue